src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/InstantiateArgumentExceptionsCorrectlyTests.cs

```diff
@@ -368,6 +593,116 @@ End Sub
+                End Class");
+        }
+
+
+        [Theory]
+        [InlineData("public", "dotnet_code_quality.api_surface = private", false)]
+        [InlineData("private", "dotnet_code_quality.api_surface = internal, public", false)]
+        [InlineData("public", "dotnet_code_quality.CA2208.api_surface = private, public", true)]
+        [InlineData("public", "dotnet_code_quality.CA2208.api_surface = internal, private", false)]
+        [InlineData("public", "dotnet_code_quality.CA2208.api_surface = Friend, Private", false)]
+        [InlineData("public", @"dotnet_code_quality.api_surface = all
+                                        dotnet_code_quality.CA2208.api_surface = private", false)]
+        [InlineData("public", "dotnet_code_quality.api_surface = public", true)]
+        [InlineData("public", "dotnet_code_quality.api_surface = internal, public", true)]
+        [InlineData("public", "dotnet_code_quality.CA2208.api_surface = public", true)]
+        [InlineData("public", "dotnet_code_quality.CA2208.api_surface = all", true)]
+        [InlineData("public", "dotnet_code_quality.CA2208.api_surface = public, private", true)]
+        [InlineData("public", @"dotnet_code_quality.api_surface = internal
+                                        dotnet_code_quality.CA2208.api_surface = public", true)]
+        public async Task EditorConfigConfiguration_ApiSurfaceOption_Test(string accessibility, string editorConfigText, bool expectDiagnostic)
+        {
+            var exception = expectDiagnostic ? @"[|new System.ArgumentNullException(""first is null"")|]" : @"new System.ArgumentNullException(""first is null"")";
+
+            await new VerifyCS.Test
+            {
+                TestState =
+                {
+                    Sources =
+                    {
+                        $@"
+public class C
+{{
+    {accessibility} void Test(string first)
+    {{
+         throw {exception};
+    }}
+}}"
+                    },
+                    AdditionalFiles = { (".editorconfig", editorConfigText) }
+                },
+                MarkupOptions = MarkupOptions.UseFirstDescriptor
+            }.RunAsync();
+
+            exception = expectDiagnostic ? @"[|New System.ArgumentNullException(""first is null"")|]" : @"New System.ArgumentNullException(""first is null"")";
+
+            await new VerifyVB.Test
+            {
+                TestState =
+                {
+                    Sources =
+                    {
+                    $@"
+ Public Class C
+    {accessibility} Sub Test(first As String)
+        Throw {exception}
+     End Sub
+ End Class"
+                    },
+                    AdditionalFiles = { (".editorconfig", editorConfigText) }
+                },
+                MarkupOptions = MarkupOptions.UseFirstDescriptor
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task EditorConfigConfiguredPublic_PrivateMethods_TriggeringOtherRules_DoesNotWarn()
```

Comment:
I think this test becomes redundant to take the approach suggested in https://github.com/dotnet/roslyn-analyzers/pull/3500/files#r413392484, which also validates that flipping the accessibility of the method from private to public on the same test generates diagnostics.

Reply:
Its not really testing that, the `MatchesConfiguredVisibility(...)` method takes DiagnosticDescriptor argument, for which i passed one of the 3 descriptors of the analyzer, i was wondering if i need ot call that method for each descriptor or it will work for all, and it is testing if the api surface option works for other descriptors even if i didn't checked that especially for those descriptors